### PR TITLE
Add PVG rule enforcing non-zero minimum_ring_size

### LINKED ISSUE #31724: Broken config for ring hash LB with minimum_ring_size of 0
Ring hash load balancer config with `minimum_ring_size` set to 0 is accepted however it results in host selection always failing.

Envoy need to have config validation that either rejects minimum_ring_size value of 0 or uses the default value of 1024 in this case.

### PR DESCRIPTION
**Pull request by antoniovleonti**
Commit Message: Add PVG rule enforcing non-zero minimum_ring_size
Additional Description: This will break any configs currently setting minimum_ring_size to zero. This shouldn't be an issue, because those configs already cause host selection to fail. See: #31724
Risk Level: Low
Testing: Unit test
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a
Fixes #31724

### CODE REVIEW

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
Limiting the range should be ok, as if one sends a config with a value of 0, it currently results in [scale](https://github.com/envoyproxy/envoy/blob/21b52ba73d8ebbb51834d529a68f55ea2ec5e614/source/extensions/load_balancing_policies/ring_hash/ring_hash_lb.cc#L160) being 0, and the ring-hash will not work.

/assign-from @envoyproxy/senior-maintainers

File: examples/shared/node/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:21.5-bookworm-slim@sha256:9df21707e27d98ee83e079fa47c5c0c5fba6c768e566643f8944b00b00afc4a2 as node-base
+FROM node:21.6-bookworm-slim@sha256:e8a7eb273ef8a9ebc03f0ad03c0fd4bbc3562ec244691e6fc37344ee2c4357d2 as node-base
```
Comment (alyssawilk) [inline examples/shared/node/Dockerfile:1]: why are these changes inhere?

**Review by alyssawilk — COMMENTED** [review]
I'm happy with API shephard sign off (I thought it wasn't allowed despite being a bugfix) but quick question about some changes below
/wait-any

File: source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h [hunk 2]
```diff
@@ -162,7 +162,7 @@ class BandwidthLimiter : public Http::StreamFilter, Logger::Loggable<Logger::Id:
   std::unique_ptr<Envoy::Extensions::HttpFilters::Common::StreamRateLimiter> response_limiter_;
   Stats::TimespanPtr request_latency_;
   Stats::TimespanPtr response_latency_;
-  std::chrono::milliseconds request_duration_;
+  std::chrono::milliseconds request_duration_ = zero_milliseconds_;
```
Comment (nezdolik) [inline source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h:165]: is this change related to the fix?

**Comment by adisuissa** [comment]
@antoniovleonti seems that couple of changes in this PR are unintentional (for some reason I see git pushes not from you on this PR).
I'm not sure exactly what's happening, but can you try to merge main. Worst case, you can open a new PR (sorry) from the new head.

File: examples/shared/node/Dockerfile (hunk 1 above)
Comment (antoniovleonti) [reply examples/shared/node/Dockerfile]: I messed up and rebased instead of merging. I believe I've fixed this (by merging). The commits are still there but the diff (in "files changed") looks correct now.

**Review by adisuissa — APPROVED** [review]
Thanks for fixing the git history.
/lgtm api

Not sure if this needs a !Google reviewer.

File: source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h (hunk 2 above)
Comment (antoniovleonti) [reply source/extensions/filters/http/bandwidth_limit/bandwidth_limit.h]: Nope, I think this is fixed.

**Review by yanavlasov — APPROVED** [review]
